q2cli/tests/test_cli.py

```diff
@@ -0,0 +1,136 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2016--, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+
+import os.path
+import unittest
+import tempfile
+
+from click.testing import CliRunner
+from qiime import Artifact, Visualization
+from qiime.core.testing.type import IntSequence1
+
+from q2cli._info import info
+from q2cli._tools import tools
+from q2cli.cli import QiimeCLI
+
+
+class CliTests(unittest.TestCase):
+
+    def setUp(self):
+        self.runner = CliRunner()
+        self.artifact1 = Artifact._from_view([0, 42, 43], IntSequence1, None)
+
+    def test_info(self):
+        result = self.runner.invoke(info)
+        self.assertEqual(result.exit_code, 0)
+        self.assertTrue(result.output.startswith('System versions'))
+        self.assertTrue('Installed plugins' in result.output)
+
+    def test_list_commands(self):
+        # top level commands, including a plugin, are present
+        qiime_cli = QiimeCLI()
+        commands = qiime_cli.list_commands(ctx=None)
+        self.assertTrue('info' in commands)
+        self.assertTrue('tools' in commands)
+        self.assertTrue('dummy-plugin' in commands)
+
+    def test_plugin_list_commands(self):
+        # plugin commands are present including method (from function),
+        # method (from markdown) and visualization
+        qiime_cli = QiimeCLI()
+        command = qiime_cli.get_command(ctx=None, name='dummy-plugin')
+        commands = command.list_commands(ctx=None)
+        self.assertTrue('split_ints' in commands)
+        self.assertTrue('split_ints_markdown' in commands)
+        self.assertTrue('mapping_viz' in commands)
+
+    def test_extract(self):
+        with tempfile.TemporaryDirectory() as output_dir:
```

Comment:
Move the `output_dir` creation and `self.artifact1.save` logic into `setUp` b/c it's used by many tests and we'll be creating temp files in a user-configured directory in the future.